platform/macos/display_server_macos.mm

```diff
@@ -3322,7 +3322,7 @@
 	[NSApp setApplicationIconImage:icon];
 }
 
-void DisplayServerMacOS::set_icon(const Ref<Image> &p_icon) {
+void DisplayServerMacOS::window_set_icon(const Ref<Image> &p_icon, WindowID p_window) {
```

Comment:
macOS have only one icon in the Dock, window title do not have an icon. Implementation probable should ignore attempts to set icon for non-main window.

Reply:
Yeah, I agree.